core/src/edu/rice/comp413fall2020grey/Location/interest/EqualityPredicate.java

```diff
@@ -0,0 +1,19 @@
+package edu.rice.comp413fall2020grey.Location.interest;
+
+import edu.rice.comp413fall2020grey.Common.metadata.ObjectMetadataField;
+
+public class EqualityPredicate<T> implements InterestPredicate {
+    final String fieldName;
+    final T value;
+
+
+    public EqualityPredicate(ObjectMetadataField field, T value) {
+        this.fieldName = field.getFieldName();
+        this.value = value;
+    }
+
+    @Override
+    public String toMongoQuery() {
+        return String.format("eq(%s, %s)", fieldName, value);
+    }
+}
```

Comment:
will all predicates sent over gRPC be in Java class format? If it's just a string that you need to send over, I would suggest letting the gRPC compiler generate these classes for you (since they will automatically deal with serialization and sending) - this is only if this is the message itself, not an internal object location way of representing it

Reply:
old code

Reply:
just realized there's an "outdated" tag at the top, will make sure to check that next pull req

Reply:
gRPC compiler will generate message class carrying string representation of Bson MongoDB query likely in Json format. The Json will be converted on the registrar end back to Bson (hopefully).

Reply:
sounds good!